### ISSUE TITLE
Add an option to not shrink glyphs with bigger advance width

### ISSUE THREAD

**Issue by fredizzimo** [neovide/neovide#1135]
Newer versions of Nerd fonts have changed the way font bearings and advance widths are calculated. This causes some icons to become much smaller than they were before.

![different sizes](https://user-images.githubusercontent.com/2427121/146944522-f4addb94-9a13-43a7-8e39-4eda50e987bb.png)

For a full discussion see this https://github.com/ryanoasis/nerd-fonts/issues/731. Note that there's no final conclusion of what to do. But I believe that Neovim can support both variants with an option.

The bigger size icon is not usually a problem with Neovim, since most plugins have an option to add an extra space after the glyph, so that there's no overlap. And I believe that's the way it worked before the Nerd font changes as well. Therefore, Neovide could render the glyphs at the real size, using the advance width of the glyph as a reference(I think), rather than scaling it down to the fixed monospaced slot. Neovide would still only advance one step, so if you don't have a space as the next character, you would get overlapping.

Some people obviously prefers strict monospacing, so this should be an optional feature toggle.

Note that I'm not fully sure that I understand everything about the issue, so the resolution might be something else.

This is related https://github.com/neovide/neovide/issues/977, but only applies to the rendering of the old versions, where the rendered width was bigger than 1 space.

Another relevant issue in the nerd fonts repo https://github.com/ryanoasis/nerd-fonts/issues/746

**Comment by fredizzimo** [comment]
Perhaps, it's better to discuss with the nerd fonts team and see what they do. This issue affects a lot of terminal emulators, and is particularly bad for nvim users, who use a lot of plugins that uses nerd font glyphs with a space following to allow for icons of up to two characters wide.

I don't understand why all the issue reports in that repository only started a couple of weeks ago though, since the issue has been there for over a year. Perhaps most people are using the release version of Nerd fonts, which still works correctly in that regard. I have also been unaffected by it myself, since I previously used a patched version of consolas, which was patched by the release version. But now when Neovide renders the font spacing correctly I wanted to switch to FiraCode, with all the latest fixes. For example the  FiraCode version in v2.1.0 of Nerdfonts has a very ugly `fl` ligature, which is fixed in later versions.

**Comment by fredizzimo** [comment]
Actually, it looks like Neovide handles the rendering correctly, but there's another problem.

I was seriously struggling debugging this, and thinking that windows somehow did not update the installed fonts, since I could not get the bigger icons even with the 2.1.0 release of the Nerd fonts, and not with my own patched fonts with the commit mentioned in the issue reverted. I tried all kinds of different things, like disabling the font caching service and clearing the font cache, but nothing worked. 

Then I found out through the Sysinternals process monitor that Neovide loads the `Mono` version of the font, even if the normal one is specified for the guifont. So uninstalling the mono variants fixed the issue, and it also works without the patches, so the issue those bug reports does not seem to affect Neovide.

But so far I don't have any idea why the mono version is loaded instead of the requested non mono one.
